### ISSUE TITLE
pipeline  Step Publish Catalog Template "GITURL" will fail to build using ip:port

### ISSUE THREAD

**Issue by MrYuanZhen** [rancher/rancher#20563]
When using the pipeline function, I need to use the Publish Catalog Template, but when my Git URL address is filled in the IP:port mode, there will be a build failure error.

#### system message

- host：CentOS 7.6
- Docker version：18.09.5-3.el7
- Rancher version：2.2.2
- Deployment method：Rancher HA

#### Reproduce step

- pipeline add a step
-  Step Type：Publish Catalog Template
-  “Git URL”：http://ip:port/uri
-  run

#### error log
fatal: could not read Username for 'http://ip:port': No such device or address

![image](https://user-images.githubusercontent.com/37647017/58608244-3b892780-82d5-11e9-8636-4d43af4f3b12.png)

**Comment by MrYuanZhen** [comment]
@wanshaoyuan 
Found and encountered the same problem

**Comment by wanshaoyuan** [comment]
@law

**Comment by wanshaoyuan** [comment]
@gitlawr

**Comment by axeal** [comment]
A Rancher customer is also encountering this issue, attempting to push a catalog template in their pipeline to a bitbucket server with host:port

**Comment by alena1108** [comment]
@gitlawr the fix is in pipeline-tools v0.1.10

**Comment by jiaqiluo** [comment]
the bug is reproduced in Rancher:v2.2.7

Steps:
- enable pipeline
- edit config to add a new step
- set the step type as `Publish Catalog Template`
- put an git url in this format: `http://ip:port/url`
- save and run

Result:
```
fatal: could not read Username for 'http://ip:port': No such device or address
```

<img width="1421" alt="Screen Shot 2019-08-07 at 4 15 55 PM" src="https://user-images.githubusercontent.com/6218999/62664203-88215c80-b92f-11e9-9f95-92c0358b60c7.png">

**Comment by jiaqiluo** [comment]
The bug fix is validated in Rancher: master-head bc3700f8

Following the same steps as the previous comment, this time the pipeline run succeeds.

<img width="1430" alt="Screen Shot 2019-08-07 at 4 28 42 PM" src="https://user-images.githubusercontent.com/6218999/62664621-1ba75d00-b931-11e9-8359-2dede3c2d93b.png">

Also confirmed that Jenkins uses the image `pipeline-tools v0.1.10` 

<img width="1470" alt="Screen Shot 2019-08-07 at 4 31 10 PM" src="https://user-images.githubusercontent.com/6218999/62664642-32e64a80-b931-11e9-9033-3b1667727b72.png">

**Comment by jiaqiluo** [comment]
The bug fix is validated in Rancher: master-head bc3700f8

Following the same steps as the previous comment, this time the pipeline run succeeds.

<img width="1430" alt="Screen Shot 2019-08-07 at 4 28 42 PM" src="https://user-images.githubusercontent.com/6218999/62664690-59a48100-b931-11e9-8669-a1e5f1f63185.png">


Also confirmed that Jenkins uses the image `pipeline-tools v0.1.10` 

<img width="1470" alt="Screen Shot 2019-08-07 at 4 31 10 PM" src="https://user-images.githubusercontent.com/6218999/62664642-32e64a80-b931-11e9-9033-3b1667727b72.png">
